R/impute_bag.R

```diff
@@ -262,8 +266,8 @@ bake.step_impute_bag <- function(object, new_data, ...) {
         rlang::warn("All predictors are missing; cannot impute")
       } else {
         pred_vals <- predict(object$models[[imp_var]], pred_data)
-        pred_vals <- cast(pred_vals, new_data[[imp_var]])
-        new_data[[imp_var]] <- vec_cast(new_data[[imp_var]], pred_vals)
+        # For an ipred bug reported on 2021-09-14:
+        pred_vals <- cast(pred_vals, object$models[[imp_var]]$y)
```

Comment:
Is this always a factor output?

If so, this could probably be as simple as:

```r
pred_vals <- factor(pred_vals, levels = levels(object$models[[imp_var]]$y))
```

That would get more at the heart of the bug, I think

Or you could wrap that in `if (is.factor(pred_vals)) { <code> }` if `pred_vals` could be numeric

Reply:
No, it could be numeric or factor. Is there are reason to get away from `cast()`?

Reply:
I guess it looks like it does the same thing. I was just thinking about the most minimal change that targets this bug, so we can safely remove it later on once it's fixed. It doesn't really matter